app/assets/javascripts/school_administrator_dashboard/discipline_dashboard/SchoolDisciplineDashboard.js

```diff
@@ -206,7 +237,7 @@ export default class SchoolDisciplineDashboard extends React.Component {
   }
 
   studentTableRows(students) {
-    const filteredStudents = this.filterStudents(students, {shouldFilterSelectedCategory: true});
+    const filteredStudents = this.state.zoomed ? this.filterToZoomedStudents(students) : this.filterStudents(students, {shouldFilterSelectedCategory: true});
```

Comment:
On this one, it's a problem if the logic is in filterStudents because that will also restrict the chart to the currently zoomed students and the chart can no longer be reset. We could work around this, of course, but doing it this way lets us use highcharts' built-in zoom feature.

Reply:
Cool, that makes sense!  For that case writing `this.filteredStudents(students, {includeInvisible: true})` might make that more explicit.